src/main/java/com/ibm/cloud/sdk/core/security/IamRequestBasedAuthenticator.java

```diff
@@ -0,0 +1,127 @@
+/**
+ * (C) Copyright IBM Corp. 2015, 2021.
```

Comment:
This is the new base class. I'll leave the initial copyright year (2015) here since this code was copied from the other file.